src/function/mtk.jl

```diff
@@ -35,11 +54,23 @@ function instantiate_function(f, x, adtype::AutoModelingToolkit, p, num_cons=0)
         hv = f.hv
     end
 
+    expr = symbolify(ModelingToolkit.Symbolics.toexpr(ModelingToolkit.equations(sys)))
+    pairs_arr = p isa SciMLBase.NullParameters ? [Symbol(_s) => Expr(:ref, :x, i) for (i,_s) in enumerate(sys.states)] : [[Symbol(_s) => Expr(:ref, :x, i) for (i,_s) in enumerate(sys.states)]..., [Symbol(_p) => p[i] for (i,_p) in enumerate(sys.ps)]...]
+    rep_pars_vals!(expr, pairs_arr)
+
     if f.cons === nothing
         cons = nothing
+        cons_exprs = nothing
     else
         cons = (θ) -> f.cons(θ, p)
         cons_sys = ModelingToolkit.modelingtoolkitize(NonlinearProblem(f.cons, x, p))
+
+        cons_eqs = ModelingToolkit.equations(cons_sys)
+        cons_exprs = map(cons_eqs) do cons_eq
+            e = symbolify(ModelingToolkit.Symbolics.toexpr(cons_eq))
+            rep_pars_vals!(e, pairs_arr)
+            return Expr(:call, :(==), e.args[2], :0)
```

Comment:
This needs to match the bounds passed to OptimizationProblem. If we don't have enough information here to add the correct bounds, we'll need to do it in `constraint_expr`.

Reply:
Yeah we don't have the information here right now hence I had to hard code to 0, I think we can either pass the constraint bounds to `instantiate_function` or as you said manipulate the expression, I'll take a stab at it

Reply:
Oh lol, you are fast! 😂